R/checkSummations.R

```diff
@@ -32,17 +33,20 @@ checkSummations <- function(mifFile, outputDirectory = ".", template = "AR6", su
 
   data <- quitte::as.quitte(mifFile, na.rm = TRUE)
 
-  summationGroups <- getSummations(summationsFile)
-
-  if (summationsFile %in% names(summationsNames())) {
-    summationsFile <- gsub(".*piamInterfaces", "piamInterfaces", summationsNames(summationsFile))
-  }
+  if (is.null(summationsFile)) {
```

Comment:
A feature you silently kill here is specifying `summationsFile = NULL` and to get asked which summation you want. I'm not sure whether anyone uses that or whether it is used somewhere deep inside `xlsx_IIASA` etc. I think not, but maybe to be safe use `summationsFile = "+"` for the automatic feature or something like that.

Reply:
Ah, thank you for spotting that! If we decide to go with this approach, i will adjust this accordingly.

Reply:
I would also fine if you did
```
if (is.null(summationsFile) && any(grepl("\\|\\++\\|", levels(q$variable))))
```
which means you only extract the summation group from the data if any `+` is actually supplied.

Reply:
I prefer setting a summationsFile name explicitly to `extractVariables` to make it easier to  understand what is going on.